test/e2e/framework/helpers/k8s/k8s_utils.go

```diff
@@ -112,6 +138,30 @@ func (k *KubeCtl) GetPod(name, namespace string) (*v1.Pod, error) {
 	return k.clientSet.CoreV1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 }
 
+func (k *KubeCtl) GetSchedulerPod() (string, error) {
+	podNameList, err := k.GetPodNamesFromNS(configmanager.YuniKornTestConfig.YkNamespace)
+	if err != nil {
+		return "", err
+	}
+	for _, podName := range podNameList {
+		if strings.Contains(podName, configmanager.YKScheduler) {
+			return podName, nil
+		}
+	}
+	return "", errors.New("YK scheduler pod not found")
+}
+
+func (k *KubeCtl) KillKubectlProcess() error {
+	cmd := exec.Command("pkill", "-SIGINT", "kubectl")
```

Comment:
I noticed the portforward has a [close](https://github.com/kubernetes/client-go/blob/2f52a105e63e9ac7affb1a0f533c62883611ba81/tools/portforward/portforward.go#L412) function, is it possible to leverage that instead of killing kubectl process? The concern is in the future, if the testing env has some other kubectl processes, they could be killed by this as a side effect.

Reply:
@yangwwei Definitely a valid concern. That function requires a `PortForwarder` and I couldn't figure out how to get an existing one. Only method I found returning that struct was [New](https://github.com/kubernetes/client-go/blob/master/tools/portforward/portforward.go#L156) which creates a new one.

Reply:
I am not sure if ginkgo has any mechanism that can be used to share context between test suites, can you check that? If not, maybe an option is to use a singleton pattern for the port forward service? What do you think?

Reply:
`KillKubectlProcess` is invoked to terminate the initial  port-forward [here](https://github.com/apache/incubator-yunikorn-k8shim/blob/master/scripts/run-e2e-tests.sh#L193-L195). For this running port-forward process, I'm not sure how to use [client-go/portforward](https://github.com/kubernetes/client-go/blob/master/tools/portforward/portforward.go) to get the `PortForwarder` instance so we can close its listeners.

We can likely use singleton pattern to maintain a `PortForwarder` once we re-run port-forwarding from this client library. But not sure how to target the main use-case of terminating the initial port-forward regardless of our ability to share context. If we have the ability to share context, can you see that being helpful somehow to get the existing `PortForwarder` (or at least its listeners) in this scenario?

Reply:
The initial port forwarding code was actually the last step of the installation, I think we can remove it, and use your latest code to create the port forward in golang code. Then we either create a shared context between test suits or use a singleton pattern to maintain one instance to do easier creating/stopping port forward. That might be a more consistent way to manage this. What do you think?